scormcloud/reportagehelper.php

```diff
@@ -0,0 +1,459 @@
+                    case 'ytd': //year to date
+                        $reporturl .= '&dateRangeType=ytd';
+                        $reporturl .= '&dateCriteria=' . $dateRangeSettings->getDateCriteria();
+                        break;
+                    //case 'p3m': ??
+                    default:
+                        break;
+                }
+            }
+
+            //Process Tags
+            $tagSettings = $widgetSettings->getTagSettings();
+            if (isset($tagSettings)) {
+                $tagSettings = $widgetSettings->getTagSettings();
+                if ($tagSettings->hasTags('learner')) {
+                    $reporturl .= '&learnerTags=' . $tagSettings->getTagString('learner');
+                    $reporturl .= '&viewLearnerTagGroups=' . $tagSettings->getViewTagString('learner');
+                }
+                if ($tagSettings->hasTags('course')) {
+                    $reporturl .= '&courseTags=' . $tagSettings->getTagString('course');
+                    $reporturl .= '&viewCourseTagGroups=' . $tagSettings->getViewTagString('course');
+                }
+                if ($tagSettings->hasTags('registration')) {
+                    $reporturl .= '&registrationTags=' . $tagSettings->getTagString('registration');
+                    $reporturl .= '&viewRegistrationTagGroups=' . $tagSettings->getViewTagString('registration');
+                }
+
+            }
+
+            //Process Comparisons
+            $comparisonSettings = $widgetSettings->getComparisonSettings();
+            if (isset($comparisonSettings)) {
+
+                $reporturl .= '&compDateRangeType=ad';
+                $reporturl .= '&compDateCriteria=launched';
+                $reporturl .= '&compLearnerTags=_all';
+                $reporturl .= '&compDateRangeType=ad';
+                $reporturl .= '&compDateCriteria=launched';
+                $reporturl .= '&compViewLearnerTagGroups=_all';
+                $reporturl .= '&compViewCourseTagGroups=_all';
+                $reporturl .= '&compViewRegistrationTagGroups=_all';
+                $reporturl .= '&compMatchDates=1';
+            }
+        }
+        //error_log("ReportUrl: ".$reporturl);
+        return $reporturl;
+    }
+
+    public function GetReportageDate()
+    {
+        $rServiceUrl = "https://cloud.scorm.com";
+        $reportageUrl = $rServiceUrl . 'Reportage/scormreports/api/getReportDate.php?appId=' . $this->_appId;
+        return $this->submitHttpPost($reportageUrl);
+    }
+
+    /**
+     * Number of seconds to wait while connecting to the server.
+     */
+    const TIMEOUT_CONNECTION = 30;
+    /**
+     * Total number of seconds to wait for a request.
+     */
+    const TIMEOUT_TOTAL = 500;
+
+    public static function submitHttpPost($url, $postParams = null, $timeout = self::TIMEOUT_TOTAL, $curlProxy = null)
+    {
+        //foreach($postParams as $key => $value)
+        //{
+        //    echo $key.'='.$value.'<br>';
+        //}
```

Comment:
There seems to be a good chunk of debugging code still commented out here. If it can be removed, that would be ideal.

Reply:
I think I have it all removed at this point.